src/olympia/abuse/models.py

```diff
@@ -261,6 +267,11 @@ def process_decision(
     def appeal(self, *, abuse_report, appeal_text, user, is_reporter):
         if not self.can_be_appealed(is_reporter=is_reporter, abuse_report=abuse_report):
             raise CantBeAppealed
+        # Author appeals can be done without an abuse report. Unfortunately
+        # though, at the moment we still need an abuse_report to call
+        # get_entity_helper(), so we grab the first one.
+        if abuse_report is None:
+            abuse_report = self.initial_abuse_report
```

Comment:
mmm, is this safe though - we also use `abuse_report` for `get_cinder_reporter` if `not user` and additionally then update the `abuse_report` instance when `is_reporter`.  From carefully reading through it does looks like `abuse_report` won't be accessed in those cases if `abuse_report` is `None` but also, imo, we're risking a bug the next time this function (or upstream) gets slightly refactored.

Maybe `self.get_entity_helper(abuse_report or self.initial_abuse_report)` so it won't be used by accident?

Reply:
I'll implement that suggestion yeah.

Ultimately it will need to be refactored to allow omitting the `abuse_report` entirely to forward reviewer decisions to Cinder that are not made off the back of an abuse report...

Reply:
Alternative proposal (on top of the PR) to make the method more robust:
```diff
diff --git a/src/olympia/abuse/models.py b/src/olympia/abuse/models.py
index 19c6832ae1..8a37e0b867 100644
--- a/src/olympia/abuse/models.py
+++ b/src/olympia/abuse/models.py
@@ -1,6 +1,7 @@
 from datetime import datetime, timedelta
 from itertools import chain
 
+from django.core.exceptions import ImproperlyConfigured
 from django.db import models
 from django.db.models import Q
 from django.db.transaction import atomic
@@ -267,17 +268,35 @@ class CinderJob(ModelBase):
     def appeal(self, *, abuse_report, appeal_text, user, is_reporter):
         if not self.can_be_appealed(is_reporter=is_reporter, abuse_report=abuse_report):
             raise CantBeAppealed
-        # Author appeals can be done without an abuse report. Unfortunately
-        # though, at the moment we still need an abuse_report to call
-        # get_entity_helper(), so we grab the first one.
-        if abuse_report is None:
-            abuse_report = self.initial_abuse_report
-        if user:
-            appealer = CinderUser(user)
+        appealer_entity = None
+        if is_reporter:
+            if not abuse_report:
+                raise ImproperlyConfigured(
+                    'CinderJob.appeal() called with is_reporter=True without an '
+                    'abuse_report'
+                )
+            if not user:
+                appealer_entity = self.get_cinder_reporter(abuse_report)
         else:
-            appealer = self.get_cinder_reporter(abuse_report)
+            if not user:
+                # If the appealer is not an original reporter, we have to
+                # provide an authenticated user that is the author of the
+                # content.
+                raise ImproperlyConfigured(
+                    'CinderJob.appeal() called with is_reporter=False without a user'
+                )
+            if not abuse_report:
+                # Author appeals can be done without an abuse report.
+                # Unfortunately though, at the moment we still need an
+                # abuse_report to call get_entity_helper(), so we grab the
+                # first one.
+                abuse_report = self.initial_abuse_report
+        if user:
+            appealer_entity = CinderUser(user)
         appeal_id = self.get_entity_helper(abuse_report).appeal(
-            decision_id=self.decision_id, appeal_text=appeal_text, appealer=appealer
+            decision_id=self.decision_id,
+            appeal_text=appeal_text,
+            appealer=appealer_entity,
         )
         with atomic():
             appeal_job, _ = self.__class__.objects.get_or_create(job_id=appeal_id)
```

What do you think ?

Reply:
It's more robust... And makes some of the hidden complexity/assumptions clear. Go for it.